### ISSUE TITLE
server: `_inc` integration test fails on CI because it is being run twice

### ISSUE THREAD

**Issue by abooij** [hasura/graphql-engine#4602]
There seems to be a test failing [here](https://app.circleci.com/pipelines/github/hasura/graphql-engine/9560/workflows/b431f78b-16a4-4b22-a7c6-ebe6bfb4f784/jobs/117601/steps).

The origin of the problem is that the relevant integration test changes data in the DB using `_inc`, and compares the new value with an expected new value. So the process is: setup -> test -> test -> teardown, rather than setup -> test -> teardown, as would be normal when no upgrades are taking place. Since the `_inc` operator is not idempotent, this causes the test to fail the second time it is being run.

In other words, an integration test is failing because the new version has triggered an integration test to be modified. The code itself seems to be functioning correctly.

**Comment by abooij** [comment]
More concretely: the logic of `run_server_upgrade_pytest` in `.circleci/server-upgrade-downgrade/run.sh` seems to be as follows:

1. The tests are run against the latest release, skipping teardown

1. The tests are run against the current build (i.e. possibly different), now skipping setup and teardown. Clearly any test that modifies the DB after setup now has the potential to fail.

1. The tests are run against the latest release, skipping setup.

I am not sure what kind of a bug this is supposed to test againts. But here is a proposal for a testing scheme that tests the setup of an old version against the tests of a new version:

1. Run *only* the setup phase of the latest release

1. Run *only* the testing phase of the current build

1. Run *only* the teardown phase of the latest release

1. If desired, repeat steps 1-3 but with the roles reversed: setup and teardown using the current build, and testing using the latest release.

This alternative scheme would have a clear "setup -> test -> teardown" pattern, avoiding any tests from being run twice.
